shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/resources/BridgeExecutor.java

```diff
@@ -54,7 +56,15 @@ public ProcessorDTO toDTO() {
         // TODO: think about removing bridgeDTO from the processorDTO and keep only bridgeId and customerId!
         processorDTO.setBridge(this.getSpec().getBridgeDTO());
         processorDTO.setName(this.getSpec().getProcessorName());
-        processorDTO.setDefinition(this.getSpec().getDefinition());
+
+        if (this.getSpec().getProcessorDefinition() != null) {
+            try {
+                processorDTO.setDefinition(new ObjectMapper().readValue(this.getSpec().getProcessorDefinition(), ProcessorDefinition.class));
```

Comment:
Since the `ProcessorDefinition` is dynamic (every action has a different structure, as well as the filters that can have a `value` or a `values` property), we need to store the `ProcessorDefinition` as a string: the custom resource has to be a defined structure, it can not have dynamic keys.